src/network/networkd-network.c

```diff
@@ -318,6 +339,7 @@ int network_load_one(Manager *manager, OrderedHashmap **networks, const char *fi
 
                 .required_for_online = true,
                 .required_operstate_for_online = LINK_OPERSTATE_RANGE_DEFAULT,
+                .activation_policy = _ACTIVATION_POLICY_INVALID,
```

Comment:
I'd like to set `ACTIVATION_POLICY_UP` here. If we do not need to detect a setting is actually specified or not, then we should set the default value here.

Reply:
the only issue with setting it directly is in the check for BindCarrier:

```
        if (!strv_isempty(network->bind_carrier)) {                                                                                                                                                                                                                           
                if (!IN_SET(network->activation_policy, _ACTIVATION_POLICY_INVALID, ACTIVATION_POLICY_BOUND))                                                                                                                                                                 
                        log_warning("%s: ActivationPolicy=bound is required with BindCarrier=. "                                                                                                                                                                              
                                    "Setting ActivationPolicy=bound.", network->filename);                                                                                                                                                                                    
                network->activation_policy = ACTIVATION_POLICY_BOUND;                                                                                                                                                                                                         

```

that currently avoids issuing the conflict warning if ActivationPolicy was unset; but if we set it to its default, we can no longer tell if it was unset or if it was set to 'up', so the warning would either need to be suppressed when someone configures both BindCarrier= and ActivationPolicy=up, or the warning would need to be logged (incorrectly) when someone configures only BindCarrier=.

Do you think we should just drop the warning completely?

Reply:
Ah, indeed. Please ignore my comment here.